Support for Spring 5

ankurpathak opened the pull request:
Fixes: gh-3736

ankurpathak commented:
Pull request for the issue:
https://github.com/eclipse-ee4j/jersey/issues/3736

jansupol commented:
Please follow the [WIKI](https://github.com/eclipse-ee4j/jersey/wiki/Missing-ECA,-Missing-Sign-off).

ankurpathak commented:
@jansupol I can't understand it becoz it uses lot of terms i have never heart and used. Can you please simplify it for this project only like steps I need to do.

jansupol commented:
@ankurpathak The wiki has it simplified. 
1) commit with -s
2) Need an Eclipse Account
3) Need an ECA signed, which is simply done when logged to the account

ankurpathak commented:
> @ankurpathak The wiki has it simplified.
> 
> 1. commit with -s
> 2. Need an Eclipse Account
> 3. Need an ECA signed, which is simply done when logged to the account

@jansupol I signed the commit.

ankurpathak commented:
@jansupol Can we have faster progress on this issue,??

jansupol commented:
The files have a wrong license. Some use CDDL/GPL instead of new EPL. The copyright year 2019 should be included, too.

Otherwise, it looks like Spring 4 dependency has just been replaced with Spring 5 and the rest is copied from spring4 module? If so, I'd keep the original starting year in a copyright header and updated only the second year. 

It looks good, but we need to file a CQ for Spring 5 first

jansupol commented:
Created
- [CQ 19278](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=19278) for spring-aop
- [CQ 19280](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=19280) for spring-beans
- [CQ 19277](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=19277) for spring-core
- [CQ 19279](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=19279) for spring-web

ankurpathak commented:
> The files have a wrong license. Some use CDDL/GPL instead of new EPL. The copyright year 2019 should be included, too.
> 
> Otherwise, it looks like Spring 4 dependency has just been replaced with Spring 5 and the rest is copied from spring4 module? If so, I'd keep the original starting year in a copyright header and updated only the second year.
> 
> It looks good, but we need to file a CQ for Spring 5 first

@jansupol I just brought back the pull request from Old Archived Repository to this repository. Thats the reason it was having issues wth licences. I have fixed the licence to new EPL with orignal start year and new end year as 2019.

ankurpathak commented:
> Created
> 
> * [CQ 19278](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=19278) for spring-aop
> * [CQ 19280](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=19280) for spring-beans
> * [CQ 19277](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=19277) for spring-core
> * [CQ 19279](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=19279) for spring-web

@jansupol What is CQ? Does I need to do any think related to it for progessof issue?

jansupol commented:
Regarding the CQs, you do not need to worry about them. They are being processed by legal now.

jansupol commented:
The CQs were approved

jansupol commented:
@ankurpathak Should the SpringComponentProvider be modified as described in #4080?

ankurpathak commented:
@jansupol @snicoll @jhoeller @rwinch @jgrandja I think we can as we have hirachy of contexts possible in Spring. But it will be a great if we take help of someone from spring project here or someone who wrote support for orignal spring4 support. So I am tagging some experts from Spring Projects here with this message. Hello Spring Team, can you please help us to resolve this fast, it will be a great help.

jansupol commented:
@ankurpathak Can you please update the PR as in #4080?

ankurpathak commented:
> @ankurpathak Can you please update the PR as in #4080?

@jansupol Modified PR to include changes of #4080.

senivam commented on ext/spring5/src/test/java/org/glassfish/jersey/server/spring/filter/TestFilter.java:
```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2014, 2018 Oracle and/or its affiliates. All rights reserved.
```
copyright year is not correct...

ankurpathak replied:
@senivam Yes. Its corrected.

senivam commented on ext/spring5/src/main/java/org/glassfish/jersey/server/spring/scope/package-info.java:
```diff
@@ -0,0 +1,21 @@
+ */
+
+/**
+ * Jersey server-side Spring 4 integration injection scopes related classes.
```
Spring 5?

also it would be great to add @ since version here and in other package-info files. Coming version is 2.29

Actually all included package-info.java files have same reference to Spring 4. I suppose it shall be corrected to Spring 5.

ankurpathak replied:
@senivam Both package-info.java corrected to Spring 5 and @ since 2.29 added.
